project_setup/skeletonize-existing-repository.md

```diff
@@ -0,0 +1,147 @@
+
+## General overview ##
+
+The general outline of how to add a skeleton to a repository is:
+
+1. [Add the skeleton as a remote](#add-a-skeleton-as-remote) to the
+non-skeletonized repository
+1. Pull with `--allow-unrelated-histories`
+1. [Fix all the inevitable conflicts](#fix-merge-conflicts)
+1. [Review non-conflicting changes](#review-non-conflicting-changes) to
+prevent merging destructive upstream changes
+1. [Update skeleton's `example` references](#update-skeletons-example-references)
+1. [Set up pre-commit](#set-up-pre-commit)
+1. Fix additional problems that may arise
+1. [Make a pull request](#make-a-pull-request)
+
+## Add a skeleton as remote ##
+
+First, decide which of the available skeletons fits your existing repository.
+To see a list of available skeletons, use the `skeleton list` command or see
+the [list of skeletons](skeleton-list.md).
+
+As an example, we'll be using [`skeleton-python-library`](https://github.com/cisagov/skeleton-python-library)
+in this document.
+
+```sh
+cd <target_repository>
+git remote add skeleton-parent git@github.com:cisagov/skeleton-python-library.git
+
+# You can verify the remote has been added by
+git remote --verbose
+
+# Create a new branch for this work
+git checkout -b skeletonize
+
+# Pull skeleton's history
+git pull skeleton-parent develop --allow-unrelated-histories
+```
+
+## Fix merge conflicts ##
+
+This merge process will almost certainly fail, resulting in merge conflicts.
+The next step is to fix those conflicts and add the files once the fixes are
+in place.
+
+```sh
+# Determine which files need fixes
+git status
+
+# After fixing the merge conflicts in a file,  add it
+git add <filename>
+
+...
+# When all conflicts have been fixed and added, commit to complete the merge
+# Remember to add a descriptive and useful commit message
+git commit
+```
+
+## Review non-conflicting changes ##
+
+You don't only have to fix merge conflicts. It is important to also look at
+the unconflicted changes listed in the output of `git status` and verify that
+you want to include all those changes.
+
+```sh
+git diff origin/develop
+```
+
+This step is often overlooked because it is rarely needed, but it can save you
+from merging in destructive upstream changes.
+
+## Update skeleton's `example` references ##
+
+This step includes such activities as:
+
+- Update `setup.py` with non-example information
+- Arrange into appropriate folders, such as `src` and `test`
+- Update the `codeowners` to reflect subject matter expertise and
+codebase familiarity
+  - Aim to have at least two codeowners for every repository
+
+Some skeletons need additional configuration, such as with
+`skeleton-python-library` and its module structure inside `src/example`.
+
+## Set up pre-commit ##
+
+The skeleton will bring along with it our standard pre-commit hook
+configurations, including linting and other checks.
+
+```sh
+# Install pre-commit
+pre-commit install
```

Comment:
They will want to have created their virtualenv before this.  We should mention both the `setup.sh` script and the instructions in `CONTRIBUTING.md` - both should be present in the repo after merging in the skeleton.

Reply:
Added in https://github.com/cisagov/development-guide/pull/32/commits/fb557ade7da9952c3bd1c47ea4c548e40d0c4dcd